apps/web/src/api/hooks/use-environment-changes.ts

```diff
@@ -1,24 +1,21 @@
 import { useQuery } from 'react-query';
+import { useEnvController } from '../../store/use-env-controller';
 import { getPromotedChanges, getUnpromotedChanges } from '../changes';
+import { QueryKeys } from '../query.keys';
 
 export function useEnvironmentChanges() {
+  const { environment } = useEnvController();
   const {
     data: changes,
     isLoading: isLoadingChanges,
     refetch: refetchChanges,
-  } = useQuery('currentChanges', getUnpromotedChanges, {
-    placeholderData: () => generateChanges(false),
-    enabled: false,
-  });
+  } = useQuery([QueryKeys.currentUnpromotedChanges, environment], getUnpromotedChanges, {});
```

Comment:
```suggestion
  } = useQuery([QueryKeys.currentUnpromotedChanges, environment?._id], getUnpromotedChanges, {});
```

Not sure if we should pass the entire object or just the id to the cache?